pvlib/test/test_pvsystem.py

```diff
@@ -148,6 +148,87 @@ def test_PVSystem_physicaliam(mocker):
     assert iam < 1.
 
 
+def test_iam_martin_ruiz():
+
+    aoi = 45.
+    a_r = 0.16
+    expected = 0.98986965
+
+    # will fail of default values change
+    iam = pvsystem.iam_martin_ruiz(aoi)
+    assert_allclose(iam, expected)
+    # will fail of parameter names change
+    iam = pvsystem.iam_martin_ruiz(aoi=aoi, a_r=a_r)
+    assert_allclose(iam, expected)
+
+    a_r = 0.18
+    aoi = [-100, -60, 0, 60, 100, np.nan, np.inf]
+    expected = [0.0, 0.9414631, 1.0, 0.9414631, 0.0, np.nan, 0.0]
+
+    with np.errstate(invalid='ignore'):
+        # check out of range of inputs as list
+        iam = pvsystem.iam_martin_ruiz(aoi, a_r)
+        assert_allclose(iam, expected, equal_nan=True)
+
+        # check out of range of inputs as array
+        iam = pvsystem.iam_martin_ruiz(np.array(aoi), a_r)
+        assert_allclose(iam, expected, equal_nan=True)
+
+        # check out of range of inputs as Series
+        iam = pvsystem.iam_martin_ruiz(pd.Series(aoi), a_r)
+        assert_allclose(iam, expected, equal_nan=True)
+
+    # check exception clause
+    with pytest.raises(RuntimeError):
+        pvsystem.iam_martin_ruiz(0.0, a_r=0.0)
+
+
+def test_iam_interp():
```

Comment:
mark with `@requires_scipy` then tests should pass.

Reply:
Thanks